applications/asset_tracker_v2/doc/data_module.rst

```diff
@@ -0,0 +1,109 @@
+.. _asset_tracker_v2_data_module:
+
+Data module
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The data module gathers data that have been sampled by other module in the system and stores it into ringbuffers.
+It keeps track of data that have been requested by the :ref:`asset_tracker_v2_app_module` and decides when data is sent cloud.
+
+.. note::
+   The data module will undergo substatial refactoring in the nearest feature. Due to this, a some of its features are not currently documented.
+
+Features
+********
+
+This section documents the various features implemented by the module.
+
+Requested data types
+====================
+
+The data module maintans a list of requested data types that is updated every time a sample request of type :c:enum:`APP_EVT_DATA_GET` is sent from the :ref:`asset_tracker_v2_app_module`.
+When all the data types present in the aforementioned list have been received by module the event :c:enum:`DATA_EVT_DATA_READY` is sent out.
+This event is a signal that all data has been gathered for a particlar sample request and that it can be encoded and sent to cloud.
+The :c:enum:`APP_EVT_DATA_GET` event also carries a sample timeout. This timeout is used by the data module to trigger a :c:enum:`DATA_EVT_DATA_READY` event, even if not all types in the requested list have been receieved.
+This acts as a failsafe in case some module that data is expected from malfunctions or does not manage to sample the requested data in time.
```

Comment:
```suggestion
This acts as a failsafe in case some modules do not manage to sample the requested data in time.
For example, it's not always possible to acquire a GNSS fix.
```